pkg/compute/rust.go

```diff
@@ -170,88 +183,57 @@ func (r Rust) Verify(out io.Writer) error {
 
 	fmt.Fprintf(out, "Found Cargo.toml at %s\n", fpath)
 
-	// 5) Verify `fastly` crate version
+	// 5) Verify `fastly` and  `fastly-sys` crate version
 	//
-	// A valid and up-to-date version of the fastly crate is required.
-	fpath, err = filepath.Abs("Cargo.lock")
-	if err != nil {
-		return fmt.Errorf("error getting Cargo.lock path: %w", err)
-	}
-
+	// A valid and up-to-date version of the fastly-sys crate is required.
 	if !common.FileExists(fpath) {
 		return fmt.Errorf("%s not found", fpath)
 	}
 
-	var lock CargoLock
-	if err := lock.Read("Cargo.lock"); err != nil {
-		return fmt.Errorf("error reading Cargo.lock file: %w", err)
+	var metadata CargoMetadata
+	if err := metadata.Read(); err != nil {
+		return fmt.Errorf("error reading cargo metadata: %w", err)
 	}
 
-	// Fetch the latest crate version from cargo.io API.
-	l, err := GetLatestCrateVersion(r.client, "fastly")
+	// Fetch the latest crate versions from cargo.io API.
+	latestFastly, err := getLatestCrateVersion(r.client, "fastly")
 	if err != nil {
 		return fmt.Errorf("error fetching latest crate version: %w", err)
 	}
-	latest, err := semver.Parse(l)
+	latestFastlySys, err := getLatestCrateVersion(r.client, "fastly-sys")
 	if err != nil {
-		return fmt.Errorf("error parsing latest crate SemVer: %w", err)
-	}
-
-	// Find the crate version declared in Cargo.lock lockfile.
-	var crate CargoPackage
-	for _, p := range lock.Package {
-		if p.Name == "fastly" {
-			crate = p
-			break
-		}
-	}
-
-	// If crate not found in lockfile, error with dual remediation steps.
-	if crate.Name == "" {
-		return errors.RemediationError{
-			Inner: fmt.Errorf("fastly crate not found"),
-			Remediation: fmt.Sprintf(
-				"To fix this error, edit %s with:\n\n\t %s\n\nAnd then run the following command:\n\n\t$ %s\n",
-				text.Bold("Cargo.toml"),
-				text.Bold(fmt.Sprintf(`fastly = "^%s"`, latest)),
-				text.Bold("cargo update -p fastly"),
-			),
-		}
+		return fmt.Errorf("error fetching latest crate version: %w", err)
 	}
 
-	// Parse lockfile version to semver.
-	version, err := semver.Parse(crate.Version)
+	fastlySysVersion, err := getCrateVersionFromMetadata(metadata, "fastly-sys")
+	// If fastly-sys crate not found, error with dual remediation steps.
 	if err != nil {
-		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+		return newCargoUpdateRemediationErr(err, latestFastly.String())
 	}
 
-	// Parse the lowest minor semver for the latest release.
-	// I.e. v0.3.2 becomes v0.3.0.
-	latestMinor, err := semver.Parse(fmt.Sprintf("%d.%d.0", latest.Major, latest.Minor))
-	if err != nil {
-		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+	// If fastly-sys version is on a lower than the latest, error with dual
+	// remediation steps.
+	if fastlySysVersion.LT(*latestFastlySys) {
```

Comment:
I tried reading the docs for this semver library and couldn't tell whether this is a strict less-than or a compatibility-respecting less-than. For example, we would want this to return false for `"0.3.4".LT("0.3.5")` but not `"0.3.4".LT("0.4.0")`

Reply:
Good catch. I've refactored to use a different semver library which supports constraint ranges. Let me know what you think.